game-app/game-core/src/main/java/games/strategy/triplea/delegate/power/calculator/AvailableSupports.java

```diff
@@ -131,9 +131,10 @@ int giveSupportToUnit(final Unit unit) {
           continue;
         }
 
-        final int numSupportAvailableToApply = getSupportAvailable(rule);
-        for (int i = 0; i < numSupportAvailableToApply; i++) {
-          final Unit supporter = getNextAvailableSupporter(rule);
+        final int numSupportAvailableToApply = getSupportersAvailable(rule);
+        for (int i = 1; i <= numSupportAvailableToApply; i++) {
+          int count = Math.min(i, getSupportersAvailable(rule));
```

Comment:
I don't understand this logic. Wouldn't `count` always be `i` in this case?

Reply:
> I don't understand this logic. Wouldn't `count` always be `i` in this case?

No, The number of supporters can be reduced by getNextAvailableSupporter. Because of this there has to check if how many supporters available with getSupportersAvailable.

Cheers...

Reply:
I see, that's very subtle. From reading the code, it's entirely not obvious that `getSupportersAvailable()` would return different values as we iterate through the loop. At the very least, that should be commented about right above this line.

But perhaps we could also find a way to make it clearer from the code itself - e.g. by renaming that function to give a hint that it's changing (e.g. `howManySupportersAreLeft(rule)` or something).